### PR TITLE
Update international IP address rule card behavior when the rule is being affected by another configuration

### LINKED ISSUE #6320: Update rule card behavior when the rule is being affected by another configuration
### Description

We should replace the rule toggle with a notice if the rule is affected by external configuration.

Image

### Step
- Go into WC core settings and set your country option to sell to all countries.
- Go to FRT advanced settings and see the notice and you'll see the toggle.

### Acceptance Criteria
- The notice is already showing, but the toggle is still visible. The toggle should be removed if we're showing the notice.

### Current rule card behavior screenshot

Screenshot on 2023-10-17 at 09_35_20

### Additional context
- p1684169738944359/1683880868.493449-slack-C04109UAJH5

### PR DESCRIPTION
**Pull request by allie500**
Fixes #6320 

#### Changes proposed in this Pull Request

This PR modifies the behavior and display of the international IP address rule card if the rule is being affected by the WooCommerce core selling locations general option. Previously, if the WooCommerce core selling locations option was set to 'sell to all countries' we'd show a notice at the bottom of the card but still display the toggle to enable the rule as shown in the screenshot below:

275852485-4e80f817-04e9-415a-ab7a-4baa484daf8f

The changes in this PR have updated the rule card display to not show the rule toggle and instead show the notice in the toggle's place with updated text indicating that filter is disabled because the WooCommerce core selling locations option is set to 'sell to all countries'. See the screenshot below:

Screenshot on 2023-12-04 at 11_35_30

The behavior and display of the rule card has not been changed if the the WooCommerce core selling locations option is set to the 'specific countries' or 'all countries except' options.

#### Testing instructions

* Check out this branch.
* All tests should pass.
* Go to WooCommerce > Settings
* Select 'Sell to all countries' in the Selling location(s) dropdown.
* Save.
* Go to Payments > Settings
* Scroll down to the fraud protection section, click advanced, then click configure.
* Scroll down to the International IP Address card and confirm that it looks like the second screenshot above. 
* Next, bring up your Jurassic Tube tunnel and confirm that the card looks okay on mobile.
* Go back to WooCommerce > Settings
* Select 'Sell to specific countries' in the Selling location(s) dropdown.
* Add a country or two to the 'Sell to specific countries' field if needed.
* Save.
* Scroll down to the fraud protection section, click advanced, then click configure.
* Scroll down to the International IP Address card and confirm that the toggle is present and that you have an info notice stating that orders from outside your specific countries will be blocked by the filter.
* Go back to WooCommerce > Settings
* Select 'Sell to all countries, except for...' in the Selling location(s) dropdown.
* Add a country or two to the 'Sell to all countries, except for...' field if needed.
* Save.
* Scroll down to the fraud protection section, click advanced, then click configure.
* Scroll down to the International IP Address card and confirm that the toggle is present and that you have an info notice stating that orders from the following countries will be blocked by the filter.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [x] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `7827` or branch name `update/6320-rule-card-behavior-if-affected-by-other-config` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/6320-rule-card-behavior-if-affected-by-other-config) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: 9228bafa09b2e0c0126b63f6699da98e7dd7bbb2
- Build time: 2023-12-07 14:23:17 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/settings/fraud-protection/advanced-settings/cards/international-ip-address.tsx [hunk 1]
```diff
@@ -11,11 +11,16 @@ import interpolateComponents from '@automattic/interpolate-components';
...
 import FraudProtectionRuleDescription from '../rule-description';
+import FraudProtectionRuleCardNotice from '../rule-card-notice';
 import FraudProtectionRuleToggle from '../rule-toggle';
 import AllowedCountriesNotice from '../allow-countries-notice';
 import { getAdminUrl } from 'wcpay/utils';
+import { getSupportedCountriesType } from '../utils';
 
 const InternationalIPAddressRuleCard: React.FC = () => {
+	const supportedCountriesType = getSupportedCountriesType();
+	const supportsAllCountries = 'all' === supportedCountriesType;
```
Comment (jessepearson) [inline client/settings/fraud-protection/advanced-settings/cards/international-ip-address.tsx]: Is there a benefit to adding the variable `supportedCountriesType` vs just doing it this way:

```
const supportsAllCountries = 'all' === getSupportedCountriesType();
```

**Review by jessepearson — CHANGES_REQUESTED** [review]
I had a question about a variable declaration that was done. I tested and it works either way, so it's not a necessary change if it's better for clarity.

There's the issue of the country names with extended ascii characters not displaying correctly. Do you think you can fix that in this PR without much hassle, or do you think it needs another issue and separate PR?

Screenshot 2023-12-05 at 11 22 27 AM

File: client/settings/fraud-protection/advanced-settings/cards/international-ip-address.tsx (hunk 1 above)
Comment (allie500) [reply client/settings/fraud-protection/advanced-settings/cards/international-ip-address.tsx]: No, there is no benefit. I think the one liner is perfectly clear and or course cleaner. Thanks for the suggestion. I've updated it in https://github.com/Automattic/woocommerce-payments/pull/7827/commits/2c9afd5ed76b4bd0952ca489a42a8815362d47e0.

**Comment by allie500** [comment]
> There's the issue of the country names with extended ascii characters not displaying correctly. Do you think you can fix that in this PR without much hassle, or do you think it needs another issue and separate PR?
> 
> Screenshot 2023-12-05 at 11 22 27 AM

Thanks for catching this! I looked in to this and it needs to be fixed in Woo core. In the [WooPayments code](https://github.com/Automattic/woocommerce-payments/blob/develop/client/settings/fraud-protection/advanced-settings/allow-countries-notice.tsx#L77) we are getting the countries from `wcSettings.countries`. 

```
...
return (
		<FraudProtectionRuleCardNotice type={ 'info' }>
			{ getNoticeText( supportedCountriesType, isBlocking ) }
			<strong>
				{ settingCountries
					.map(
						( countryCode ) =>
							wcSettings.countries[ countryCode ] ?? false
					)
					.filter( ( element ) => element )
					.join( ', ' ) }
			</strong>
		</FraudProtectionRuleCardNotice>
	);
...
```

If you run `wcSettings.countries['ST']` in the browser console, you get the following value returned:
`'S&atilde;o Tom&eacute; and Pr&iacute;ncipe'`

That is hardcoded in the core countries file:
https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/i18n/countries.php#L208

I can open an issue in core to track this.

**Comment by allie500** [comment]
> If you run `wcSettings.countries['ST']` in the browser console, you get the following value returned: `'S&atilde;o Tom&eacute; and Pr&iacute;ncipe'`
> 
> That is hardcoded in the core countries file: https://github.com/woocommerce/woocommerce/blob/trunk/plugins/woocommerce/i18n/countries.php#L208
> 
> I can open an issue in core to track this.

Actually, it does look like something on our end as core does properly decode the entities on their end. I'll open a separate issue for it in this repo.

**Review by jessepearson — APPROVED** [review]
Thanks for updating the variable declaration and opening the issue related to the country display. This is good to go 😁

**Pull request merged**